src/KubernetesClient.Models/KubernetesJson.cs

```diff
@@ -92,6 +92,10 @@ public static TValue Deserialize<TValue>(Stream json, JsonSerializerOptions json
             return JsonSerializer.Deserialize<TValue>(json, jsonSerializerOptions ?? JsonSerializerOptions);
         }
 
+        public static TValue Deserialize<TValue>(JsonElement json, JsonSerializerOptions jsonSerializerOptions = null)
```

Comment:
is it ok to make this internal?

Reply:
IMHO this should be kept public, otherwise it will not be possible to make use of it when using `ICustomObjectsOperations`.

Reply:
no worried, it by default
```
[assembly: InternalsVisibleTo("KubernetesClient")]
```

this api looked unreasonable to people who do not have context. so let keep it secret

Reply:
To clarify:

Lets suppose in my application I want to read some custom object, I'am still forced to use double serialization/deserialization without this being public:

```csharp
var result = await kubernetes.CustomObjects.ListClusterCustomObjectAsync(/*...*/);
var customObj = KubernetesJson.Deserialize<V1MyCustomObj>(result.ToString());
```

a possible work around would be to cast the result to `JsonElement` and invoke `Deserialize` on it:

```csharp
var result = await kubernetes.CustomObjects.ListClusterCustomObjectAsync(/*...*/);
var customObj = ((JSonElement)result).Deserialize<V1MyCustomObj>();
```

sadly this would not use the custom convertes registered in `KubernetesJson`.

Another way would be to make `JsonSerializerOptions` in `KubernetesJson` public and make the new `Deserialize` method internal, which would enable app's to use the API this way:

```csharp
var result = await kubernetes.CustomObjects.ListClusterCustomObjectAsync(/*...*/);
var customObj = ((JSonElement)result).Deserialize<V1MyCustomObj>(KubernetesJson.JsonSerializerOptions);
```

Reply:
@tg123 which option do you prefer ?

Reply:
I've changed the code generation to generate methods for CustomObjects to include a type parameter.
IMHO `GenericClient` could be marked as obsolete.

Reply:
@tg123 Can you please review again? Thanks!

Reply:
This is awesome, i will take a deeper look after runrise Thanks